### PR TITLE
Alerting: Fix Loki buildinfo request error

### LINKED ISSUE #48627: Alerting: Error on loading rules from Loki
**What happened**:
Navigating to alert rules throw the error:
```
Errors loading rules
Failed to load the data source configuration for [Loki](https://grafana.signintra.com/datasources/edit/P8E80F9AEF21F6940): Cannot destructure property 'features' of 'l.data.data' as it is undefined.
```

**How to reproduce it (as minimally and precisely as possible)**:
add a Loki datasource with an active ruler and configure alerting

**Anything else we need to know?**:
It works fine for Mimir because the `features` property is exposed: https://github.com/grafana/mimir/blob/90aeee02de04199e0f0e4d981ce2460478f76392/pkg/util/version/info_handler.go#L22

In Loki it's not: https://github.com/grafana/loki/blob/8b15632dd8e584c04facd48a1cbe26aa807516e3/pkg/loki/version_handler.go#L14

It also works with the same setup for Loki, if I downgrade Grafana to 8.3.5 or 8.4.6, but fails with for 8.5.0

Loki DS definition:
```json
{
  name: 'Loki',
  type: 'loki',
  access: 'proxy',
  // BEGIN TODO: OPSORG
  orgId: std.parseInt(opsid),
  // END TODO: OPSORG
  url: 'https://' + $.config.lokiHost,
  version: 1,
  editable: false,
  jsonData: {
    serverName: $.config.lokiHost,
    tlsAuth: true,
    tlsAuthWithCACert: true,
  },
  secureJsonData:
    {
      tlsCACert: loki.issuing_ca,
      tlsClientCert: loki.certificate,
      tlsClientKey: loki.private_key,
    },
},
```

**Environment**:
- Grafana version: 3.5.1
- Data source type & version: Loki 2.5.0

### PR DESCRIPTION
**Pull request by konrad147**
**What this PR does / why we need it**:
This PR prevents Loki-based data sources from fetching `api/v1/status/buildinfo` because right now, the response doesn't provide any helpful information.

**Which issue(s) this PR fixes**:
Fixes #48627, #48938

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by gillesdemey** [comment]
Does this supersede https://github.com/grafana/grafana/pull/48924 ?

**Comment by konrad147** [comment]
> Does this supersede #48924 ?

Yes

File: public/app/features/alerting/unified/api/buildInfo.test.ts [hunk 1]
```diff
@@ -36,7 +36,11 @@ describe('buildInfo', () => {
         })
       );
 
-      const response = await fetchDataSourceBuildInfo({ url: '/datasource/proxy', name: 'Prometheus' });
+      const response = await discoverDataSourceFeatures({
+        url: '/datasource/proxy',
+        name: 'Prometheus',
+        type: 'prometheus',
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/api/buildInfo.test.ts:42]: Do we have an `enum` for this? I'd love for the compiler to tell us when we're sending an unknown query string value here

File: public/app/features/alerting/unified/api/buildInfo.ts [hunk 2]
```diff
@@ -18,27 +18,25 @@ import { fetchTestRulerRulesGroup } from './ruler';
...
-      showSuccessAlert: false,
-    })
-  ).catch((e) => {
-    if ('status' in e && e.status === 404) {
-      return null; // Cortex does not support buildinfo endpoint, we return an empty response
-    }
+export async function discoverDataSourceFeatures(dsSettings: {
+  url: string;
+  name: string;
+  type: string;
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/api/buildInfo.ts:24]: See my comment on typing earlier

**Review by gillesdemey — APPROVED** [review]
Looks great! 🤩 

EDIT: failing tests in the enterprise branch seem unrelated to our changes

**Review by joeblubaugh — APPROVED** [review]
Thanks for a much better PR than I wrote! :-) 

I also really like a lot of the action renaming here - discoverFeatures is really clear naming!

**Pull request merged**